src/pages/index.js

```diff
@@ -21,91 +34,155 @@ popUpImageModal.setEventListeners();
+avatarPopUp.setEventListeners();
+//Buttons that Open Popup with forms
+// grab from userinfo name + descriptions
+const newUserInfo = new UserInfo(constants.selectors);
+constants.profileEditButton.addEventListener("click", () => {
+  handleFormFill(newUserInfo.getUserInfo());
+  return profilePopup.openModal();
+});
 constants.addButton.addEventListener("click", () => {
-  addFormValidator.toggleButtonState();
   return cardPopUp.openModal();
 });
+
+constants.profileAvatar.addEventListener("click", () => {
+  return avatarPopUp.openModal();
+});
+
 //initializes new section renders inittial cards and new ones
-const cardSection = new Section(
-  {
-    items: constants.initialCards,
-    renderer: renderCard,
-  },
-  constants.selectors.cardSection
-);
-cardSection.renderItems(constants.initialCards);
+let cardSection;
+api.getInitialCards().then((data) => {
+  cardSection = new Section(
+    {
+      items: data,
+      renderer: renderCard,
+    },
+    constants.selectors.cardSection
+  );
+  cardSection.renderItems();
+});
+
 //Submit Button handler
+//Toggle Submit button State
 /**Event Handlers*/
-function handleProfileEditSubmit(modalInputs) {
-  newUserInfo.setUserInfo(modalInputs.title, modalInputs.description);
-  profilePopup.closeModal();
+function handleProfileEditSubmit(modalInputs, popUpForm) {
+  api
+    .updateInfo(modalInputs)
+    .then(() => {
+      const { title, description } = modalInputs;
+      newUserInfo.setUserInfo(title, description);
+      editFormValidator.toggleButtonState();
+    })
+    .finally(() => {
+      popUpForm.querySelector(".modal__form-button").innerText = "Save";
+      profilePopup.closeModal();
+    });
 }
-function handleAddModalSubmit(modalInputs) {
+function handleAddModalSubmit(modalInputs, popUpForm) {
   const name = modalInputs.place;
   const link = modalInputs.Url;
-  //const card = new Card({ name, link }, "#card-template");
-  const newCard = renderCard({ name, link });
-  cardSection.addItems(newCard);
-
-  cardPopUp.closeModal();
+  
+  api
+    .postCards(modalInputs)
+    .then((res) => {
+      const newCard = renderCard(res);
+      cardSection.addItems(newCard);
+      popUpForm.reset();
+      addFormValidator.toggleButtonState();
+    })
+    .finally(() => {
+      popUpForm.querySelector(".modal__form-button").innerText = "Save";
+      cardPopUp.closeModal();
+    });
 }
+function handleAvatarSubmit(modalInputs, popUpForm) {
+  api
+    .updateAvatar(modalInputs)
+    .then(() => {
+      const link = modalInputs.Url;
+      constants.profileAvatar.src = link;
+      newUserInfo.setAvatar(link);
+      popUpForm.reset();
+      avatarFormValidator.toggleButtonState();
+    })
+    .finally(() => {
+      popUpForm.querySelector(".modal__form-button").innerText = "Save";
+      avatarPopUp.closeModal();
+    });
+}
+api.loadInfo().then((data) => {
+  const { name, about, avatar } = data;
+  newUserInfo.setUserInfo(name, about);
+  constants.profileAvatar.src = avatar;
```

Comment:
`      constants.profileAvatar.src = avatar;
`

Please, delete it

`newUserInfo.setAvatar(avatar);` should set the avatar